modules/imgcodecs/src/grfmt_jpegxl.cpp

```diff
@@ -76,7 +89,7 @@ ImageDecoder JpegXLDecoder::newDecoder() const
     return makePtr<JpegXLDecoder>();
 }
 
-bool JpegXLDecoder::read(Mat* pimg)
+bool JpegXLDecoder::read(Mat& img)
```

Comment:
The function behaviour depends on emptiness of the input Mat. For now it's not true, even if user mat is empty and allocated by caller side. IMHO the approach is dangerous and may be broken during refactoring or some redesign. I propose to split the concerns and introduce bool flag. It's almost free and makes the code more readable.

Reply:
Yes, I agree with you. And I refactor to remove argument variable for read() function.

There are many changes, but I think it makes easier to fix and to modify.